src/System.Net.HttpListener/tests/GetContextHelper.cs

```diff
@@ -22,30 +22,23 @@ public GetContextHelper(HttpListener listener, string listeningUrl)
             _client = new HttpClient();
         }
 
-        public async Task<HttpListenerResponse> GetResponse()
+        public HttpListenerResponse GetResponse()
         {
             // We need to create a mock request to give the HttpListener a context.
             Task<string> clientTask = _client.GetStringAsync(_listeningUrl);
-            HttpListenerContext context = await _listener.GetContextAsync();
-            return context.Response;
+            return _listener.GetContext().Response;
```

Comment:
We should not be adding blocking; this kind of thing is why I had to make the tests run serialized rather than in parallel, as this leads to deadlocks when limiting the number of synchronization contexts available.